lib/grammars.coffee

```diff
@@ -134,6 +134,22 @@ module.exports =
       command: "julia"
       args: (context) -> [context.filepath]
 
+  Kotlin:
+    "Selection Based":
+      command: "bash"
+      args: (context) ->
+        code = context.getCode(true)
+        tmpFile = GrammarUtils.createTempFileWithCode(code, ".kt")
+        jarName = context.filename.replace(/\.kt$/g, "") + ".jar"
```

Comment:
Two thoughts here and in the other `jarName` assignment:

1. We don't need the `g` "global" replace flag since only one match for `/\.kt$` will be found (at the end of the line).
2. Instead of replacing `.kt` with an empty string replace it with `.jar` vs. appending that later.